workloads/api_load/api-load-crd.yaml

```diff
@@ -0,0 +1,26 @@
+---
+apiVersion: ripsaw.cloudbulldozer.io/v1alpha1
+kind: Benchmark
+metadata:
+  name: api-load-${WORKLOAD}-${UUID}
+  namespace: benchmark-operator
+spec:
+  uuid: ${UUID}
+  elasticsearch:
+    url: ${ES_SERVER}
+    index_name: "api-load-$(UUID}"
+  workload:
+    name: api_load
+    args:
+      gateway_url: ${GATEWAY_URL}
+      ocm_token: ${OCM_TOKEN}
+      duration: ${DURATION}
+      rate: ${RATE}
+      output_path: ${OUTPUT_PATH}
+      aws_access_key: ${AWS_ACCESS_KEY}
+      aws_access_secret: ${AWS_ACCESS_SECRET}
+      aws_account_id: ${AWS_ACCOUNT_ID}
+      cooldown: ${COOLDOWN}
+      sleep: ${SLEEP}
+      test_list: 
+       ${WORKLOAD}:
```

Comment:
Excuse my ignorance, Is it just a string, without dict values ? 
how to run a list of workloads ?

Reply:
Looks like the CR lets you run only workload then? This seems like the operator implementation limitation

Reply:
Probably not running complex scenario, but it iterates them in code https://github.com/cloud-bulldozer/benchmark-operator/blob/master/roles/api_load/templates/api_load.yml#L65.

Reply:
benchmark-operator let's you run multiple at a time

Reply:
Each test (like this example config https://github.com/cloud-bulldozer/ocm-api-load/blob/main/config.example.yaml#L28 ) can have different values for "rate" and "duration" options. 
Benchmark-operator also allows this configuration https://github.com/cloud-bulldozer/benchmark-operator/blob/master/roles/api_load/templates/api_load.yml#L71
As e2e accepts only environment variables, I need to define a new env variable for each test's rate and duration options, for example,
ACCESS_TOKEN_RATE: 1000/h
ACCESS_TOKEN_DURATION: 1
LIST_SUBSCRIPTIONS_RATE: 2000/h
LIST_SUBSCIPTION_DURATION: 1
...
CERTIFICATES_RATE: 2/s
CERTIFICATES_DURATION: 1

Please let me know If we have any other way to get these values from user.

Reply:
Currently we need to pass many environment variables to ./run.sh for invoking the test i.e
WORKLOAD=list-clusters GATEWAY_URL="http://localhost:8080" OCM_TOKEN="notARealToken" RATE=10/s AWS_ACCESS_KEY="empty" AWS_ACCESS_SECRET="empty" AWS_ACCOUNT_ID="empty" ./run.sh

instead I am thinking user can directly pass the config yaml file (i.e CONFIG=~/config.yaml ./run.sh) which run.sh parses and retrieve these variables. Another advantage is - yaml config file can define multiple tests with args
( https://github.com/cloud-bulldozer/ocm-api-load/blob/main/config.example.yaml#L31 )
 helping us run multiple tests with single invocation of run.sh

Reply:
you could also set all these variables in `env.sh` file one time and just `./run.sh`  right.
And we already dealing such scenarios in some e2e scripts and [airflow](https://github.com/cloud-bulldozer/airflow-kubernetes/blob/master/dags/openshift_nightlies/tasks/benchmarks/e2e.py) takes care of setting those secrets/vars during the task execution. IMO passing env var is fine.

Reply:
And `env.sh` file doesn't sound bad, probably tricky to create the first time, but once it's there, it would be just replacing stuff.

Reply:
Please check latest approach